source/agora/consensus/protocol/Nominator.d

```diff
@@ -624,19 +859,86 @@ extern(D):
 
         // enrollment data may be empty, but not transaction set
         if (data.tx_set.length == 0)
-            assert(0, format!"Transaction set empty for slot %u"(slot_idx));
+            assert(0, format!"Transaction set empty for slot %s"(height));
 
-        log.info("Externalized consensus data set at {}: {}", slot_idx, data);
+        log.info("Externalized consensus data set at {}: {}", height, prettify(data));
         try
         {
-            if (!this.ledger.onExternalized(data))
-                assert(0);
+            auto blocks = this.ledger.getBlocksFrom(Height(slot_idx - 1));
+            assert(!blocks.empty);  // should not happen
+            const prev_block = blocks.front;
+
+            const block = makeNewBlock(prev_block, data.tx_set, data.enrolls);
+
+            const signed_block = updateMultiSignature(block);
+
+            // If we did not sign yet then add signature
+            if (this.schnorr_pair.V !in this.slot_sigs[height])
+            {
+                log.trace("ADD BLOCK SIG at height {} for this node {}", height, this.public_key);
+                this.slot_sigs[height][this.schnorr_pair.V] = createBlockSignature(block);
+            }
+            if (signed_block == Block.init)
+            {
+                // Do not externalize if block was not signed
+                return;
+            }
+            if (!this.ledger.acceptBlock(signed_block))
+            {
+                log.error("Block was not accepted by node {}", this.public_key);
+                assert(0, format!"Block was not accepted"());
+            }
         }
         catch (Exception exc)
         {
             log.fatal("Externalization of SCP data failed: {}", exc);
             abort();
         }
+        gossipBlockSignature(ValidatorBlockSig(height, this.public_key,
+            this.slot_sigs[height][this.schnorr_pair.V]));
+    }
+
+    /// function for gossip of block sig which can be overriden in byzantine unit tests
+    extern(D) protected void gossipBlockSignature(ValidatorBlockSig block_sig) nothrow
+    {
+        // Send to other nodes in the network
+        this.network.gossipBlockSignature(block_sig);
+    }
+
+    /// Combine the received block signatures into a Schnorr multisig and check
+    /// that more than half of the validators have signed
```

Comment:
Where is the check that more than half have signed?

Reply:
This PR was intended to just add signatures but not enforce them yet.

Reply:
But I left the comments by mistake so will update them.

Reply:
Thanks, that was my assumption as well.